content/working_with/console/widgets/blueprints.md

```diff
@@ -53,6 +53,48 @@ There are also action buttons to upload a blueprint, create deployment, delete b
 The default visibility is "Tenant", and according to the logged-in user's permissions you can also choose other levels of [resource visibilities]({{< relref "working_with/manager/resource-visibility.md" >}}).<br>
 6. Click **Upload**.
 
+##### Using Terraform module
+1. Click the **Upload** button.
+2. In the menu, click **Upload from Terraform module** option.
+3. **Create blueprint from Terraform** modal will appear
+4. Enter the `Blueprint name`   
+   `Blueprint name` is the name with which you want to identify this blueprint once uploaded
+5. Select `Terraform version` (as a default, one of the terraform versions will be selected)
+6. Provide `Blueprint main information`, which consists of:
+   - `URL to a zip archive that contains the Terraform module`,
+   - `Terraform folder in the archive` - it's a selectable field from which you'll be able to select directories found inside the linked zip file.<br />
+   By default this field is disabled, until the `URL to a zip archive that contains the Terraform module` is being provided.
+   - (Optional) Credentials required for accessing `URL to a zip archive that contains the Terraform module`.
+7. (Optional) Add `Variables`, by:   
+   1. Clicking the `Variables` dropdown
+   2. Clicking the **Add** button
+   3. Filling the row fields:
+      - `Variable` - name of the variable
+      - `Source` - type of the variable, which may be selected from a dropdown<br />
+         Currently available values are: `Secret`, `Input`, `Static`.
+      - `Value / Secret key / Input name` - value of the variable (related to the selected `Source`)<br />
+         If we have not selected `Secret` `Source`, then we are able to type our own values.<br />
+         Upon selecting the `Secret` `Source` we are getting the ability to select one of the stored secrets.
+8. (Optional) Add `Environment variables`, by:   
+   1. Clicking the `Environment variables` dropdown
+   2. Clicking the **Add** button
+   3. Filling the row fields:
+      - `Variable` - name of the variable
+      - `Source` - type of the variable, which may be selected from a dropdown<br />
+         Currently available values are: `Secret`, `Input`, `Static`.
+      - `Value / Secret key / Input name` - value of the variable (related to the selected `Source`)<br />
+         If we have not selected `Secret` `Source`, then we are able to type values.<br />
+         Upon selecting the `Secret` `Source` we are getting the ability to select one of the stored secrets.
```

Comment:
As `Variables` and `Environment variables` are different only by name I think we should not duplicate the description.

Reply:
Which description are you referring to?
The one related to the `Filling the row fields` or the whole instruction about how to add an environment variable?

Reply:
As the discussion is being continued in a different [thread](https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2077#discussion_r792475201), I'm resolving this one

Reply:
Changes has been made according to the suggestion from the [continuation](https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2077#discussion_r792475201) of this discussion

Because of that, I'm resolving this conversation